frontend/src/views/Join.tsx

```diff
@@ -89,6 +88,13 @@ function JoinGamePage() {
       }).catch((err) => reject(err));
   });
 
+  // Get URL query params to determine if the roomId is provided.
+  const urlParams = new URLSearchParams(window.location.search);
+  const roomIdQueryParam = urlParams.get('room') || '';
+  if (!roomId && roomIdQueryParam && isValidRoomId(roomIdQueryParam)) {
```

Comment:
I would move this back into the useEffect block, as right now, if you delete all the characters in the input box, it auto-fills the roomId back to the whatever the query param is.

Reply:
Wow good catch, I did not notice that. The only concern though is that I'd have to add the `roomId` as a dependency to this, which would defeat this purpose. I removed that portion of the if statement and just passed in `isValidRoomId` and `setRoomId` as the dependencies, as it will just run that then on initial page load.

Since the `roomId` (I believe) will never have a value on initial page load, that check wasn't needed in the first place now that I have the `useEffect` hook.

This is different than what you suggested @alankbi, so let me know what you think.

Reply:
Yep good catch on the roomId dependency!